flutter/flutter
# Implement barrierDismissible for `showCupertinoDialog`

## Description

Adds support for dismissing cupertino dialog with barrier behavior. By default the dialog is not dismissible with a tap on the barrier. The localization files updates are based on the existing material ones.

## Related Issues

Fixes #45744

## Tests

I added the following tests:

- check if barrierDismissible is disabled by default
- check if barrierDismissible is enabled if enabled explicitly
- check for existence of the localized modalBarrierDismissLabel for cupertino

## Checklist
